packages/apollo-engine-reporting/src/durationHistogram.ts

```diff
@@ -38,8 +38,20 @@ export class DurationHistogram {
     this.incrementBucket(DurationHistogram.durationToBucket(durationNs));
   }
 
-  public incrementBucket(bucket: number) {
-    this.buckets[bucket]++;
+  public incrementBucket(bucket: number, value = 1) {
+    if (bucket >= DurationHistogram.BUCKET_COUNT) {
+      // Since we don't have fixed size arrays I'd rather throw the error manually
+      throw Error("Bucket is out of bounds of the buckets array");
+    }
+
+    // Extend the array if we haven't gotten it long enough to handle the new bucket
+    if (bucket >= this.buckets.length) {
+      let tempBufferLength = bucket - this.buckets.length + 1;
```

Comment:
I think you can do `this.buckets.length = bucket + 1; this.buckets.fill(0, oldLength)` instead of making a temporary array and then pushing it

Reply:
Didnt' know that was possible. Fixed it.